test/test_utils.h

```diff
@@ -156,6 +156,20 @@ namespace detail {
 
 namespace test_utils {
 
+	// Pin the benchmark threads (even in absence of a runtime) for more consistent results
+	struct benchmark_thread_pinner {
+		benchmark_thread_pinner() {
+			const detail::thread_pinning::runtime_configuration cfg{
+			    .enabled = true,
+			    .use_backend_device_submission_threads = false,
+			};
+			m_thread_pinner.emplace(cfg);
+			detail::thread_pinning::pin_this_thread(detail::thread_pinning::thread_type::application);
+		}
+
+		std::optional<detail::thread_pinning::thread_pinner> m_thread_pinner;
```

Comment:
```suggestion
		benchmark_thread_pinner() {
			detail::thread_pinning::pin_this_thread(detail::thread_pinning::thread_type::application);
		}

		detail::thread_pinning::thread_pinner m_thread_pinner{{
			.enabled = true,
			.use_backend_device_submission_threads = false,
		}};
```
Just out of curiosity, can designated initializers be used in this fashion?